blockreassurance.php

```diff
@@ -19,339 +19,418 @@
+            $this->css_path.'menu.css',
+        );
+
+        $aJs = array(
+            $this->lib_path.'/pickr/js/pickr.js',
+            $this->js_path.'/appearance/colorpicker.js',
+            $this->js_path.'back.js',
+            $this->js_path.'menu.js',
+            $this->js_path.'vue.min.js',
+        );
+
+        $aCss[] = '//fonts.googleapis.com/icon?family=Material+Icons';
+
+        $this->context->controller->addCSS($aCss, 'all');
+        $this->context->controller->addJS($aJs);
+        $controller->addJqueryPlugin('colorpicker');
+        $controller->addJqueryUI('ui.sortable');
     }
 
-    public function addToDB()
+    /**
+     * Check if folder img_perso is writable and executable
+     *
+     * @return void
+     */
+    private function folderUploadFilesHasGoodRights()
     {
-        if (isset($_POST['nbblocks'])) {
-            for ($i = 1; $i <= (int)$_POST['nbblocks']; $i++) {
-                $filename = explode('.', $_FILES['info'.$i.'_file']['name']);
-                if (isset($_FILES['info'.$i.'_file']) && isset($_FILES['info'.$i.'_file']['tmp_name']) && !empty($_FILES['info'.$i.'_file']['tmp_name'])) {
-                    if ($error = ImageManager::validateUpload($_FILES['info'.$i.'_file'])) {
-                        return false;
-                    } elseif (!($tmpName = tempnam(_PS_TMP_IMG_DIR_, 'PS')) || !move_uploaded_file($_FILES['info'.$i.'_file']['tmp_name'], $tmpName)) {
-                        return false;
-                    } elseif (!ImageManager::resize($tmpName, dirname(__FILE__).'/img/'.$filename[0].'.jpg')) {
-                        return false;
-                    }
-                    unlink($tmpName);
-                }
-                Db::getInstance()->execute('INSERT INTO `'._DB_PREFIX_.'reassurance` (`filename`,`text`)
-                                            VALUES ("'.((isset($filename[0]) && $filename[0] != '') ? pSQL($filename[0]) : '').
-                    '", "'.((isset($_POST['info'.$i.'_text']) && $_POST['info'.$i.'_text'] != '') ? pSQL($_POST['info'.$i.'_text']) : '').'")');
-            }
-            return true;
-        } else {
-            return false;
-        }
+        return is_writable($this->folder_file_upload)
+            && is_executable($this->folder_file_upload);
     }
 
-    public function removeFromDB()
+    /**
+     * getContent
+     *
+     *
+     * @return template
+    */
+    public function getContent()
     {
-        $dir = opendir(dirname(__FILE__).'/img');
-        while (false !== ($file = readdir($dir))) {
-            $path = dirname(__FILE__).'/img/'.$file;
-            if ($file != '..' && $file != '.' && !is_dir($file)) {
-                unlink($path);
-            }
+        $this->loadAsset();
+
+        $id_lang = $this->context->language->id;
+        $currentPage = Tools::getValue('page');
+
+        // get current page
+        if ($currentPage != 'global') {
+             $currentPage = 'global';
```

Comment:
Euh 🤔this statement looks strange

1) If `$currentPage` is equal to "global", this if statement is skipped
2) If `$currentPage` is not equal to "global", then `$currentPage` is assigned the value "global"

So what's the difference with
```
$currentPage = 'global';
```

Reply:
my bad, it was a mistake.